### PR TITLE
Feature/parallel tests

### PR DESCRIPTION
**Pull request by pogobanane**
With warm caches test time is reduced from ~29s to 9s. Tests don't run `cargo run` but the executable `./target/...` directly.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -16,6 +16,13 @@
 
 sys.path.append(str(TEST_ROOT.parent))
 
+build_artifacts = Path("/dev/null")  # folder with cargo-built executables
+
+
+def cargo_build() -> Path:
+    subprocess.run(["cargo", "build"])
+    return Path("target/debug")
```
Comment (Mic92) [inline tests/conftest.py:24]: Does this create a new build directory in ./tests?

```suggestion
    return TEST_ROOT.parent.joinpath("target", "debug")
```

`cwd=TEST_ROOT.parent` in the build command might help as well.

Comment (pogobanane) [reply tests/conftest.py:24]: `cargo build` builds into vmsh/target/debug when called from somewhere within `vmsh/`.

Comment (Mic92) [reply tests/conftest.py]: You are probably right. Anyway does not hurt to hard-code it.

**Comment by pogobanane** [comment]
dontmerge. #85 is flawed

**Pull request merged**